### ISSUE TITLE
개인별 모델 baseline 구축 및 실험 결과 공유_선민

### ISSUE THREAD

**Issue by nimnusmik** [AIBootcamp14/upstageailab-ml-competition-ml-5#75]
### ✅ Preprocessing
- [x] 결측치 제거 및 확인
- [x] '계약요일' 변수 생성
- [x] '브랜드등급' 수치화
- [x] '법정동' Label Encoding
- [x] ColumnTransformer 사용하여 수치형/범주형 처리

### ✅ Modeling
- [x] Pipeline 구성 (`preprocessor` + `RandomForestRegressor`)
- [x] train/validation split 후 학습
- [x] RMSE: 6767.60, R²: 0.9783 (local)

### ✅ KFold cross-validation 
 --- K-Fold Cross-Validation 결과 요약 ---
평균 RMSE: 6691.2896 (표준편차: 154.2763)
평균 R^2: 0.9787 (표준편차: 0.0009)

### ✅ Submission
- [x] test 데이터에 pipeline 적용하여 예측
- [x] sample_submission 파일 포맷에 맞춰 저장

### ❌ 실제 평가 점수: 73209.4350 
[제출 현황](https://stages.ai/en/competitions/363/leaderboard/list)

---
### 🖼️ 인코딩 후 시각화
<img width="1189" height="790" alt="Image" src="https://github.com/user-attachments/assets/fe5ff631-2eeb-4da8-b15e-026b9dc5796a" />

## 🔍 분석 및 추정 원인

- test 데이터 분포가 train과 달라서 generalization 실패
- encoding/전처리 시 train 기준 처리로 test에는 unseen값 발생 가능
- RandomForest는 overfitting 위험이 있어서 validation 성능만 좋았을 수 있음

---

## 💡 개선 아이디어

- [x] feature selection -> RandomForest 
- [x] LightGBM/XGBoost 등 다른 모델 실험

---
## 참고파일 
04.FeatureEngineering.py
